src/api/spec/features/webui/requests/diff_comments_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'browser_helper'
+
+RSpec.describe 'Comments with diff', :js, :vcr do
+  let(:admin) { create(:admin_user, login: 'Admin') }
+  let(:target_project) { create(:project, name: 'target_project') }
+  let(:source_project) { create(:project, :as_submission_source, name: 'source_project') }
+  let(:target_package) { create(:package, name: 'target_package', project: target_project) }
+  let(:source_package) do
+    create(:package_with_files,
+           name: 'package_a',
+           project: source_project,
+           changes_file_content: '- Fixes ------')
+  end
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           target_package: target_package,
+           source_package: source_package)
+  end
+
+  let!(:comment) { create(:comment, commentable: bs_request.bs_request_actions.first, diff_ref: 'diff_0_n1', user: admin) }
+
+  context 'reply comment' do
+    describe 'when under the beta program' do
+      before do
+        Flipper.enable(:request_show_redesign, admin)
+        login admin
+        visit request_show_path(bs_request)
+      end
+
+      it 'displays the comment in the timeline' do
+        expect(page).to have_text(comment.body)
+        expect(page).to have_text('target_project/target_package > package_a.changes')
+      end
```

Comment:
I think this context is trying to do too much: either you move this into it's own context (like "rendering comments") so you can free the before block to prepare for this test (that's why you have to cram everything un the `it` block below) or you remove this expectation and move the clicking in the before block above